pyicloud/services/photos.py

```diff
@@ -226,10 +220,71 @@ def all(self):
         return self.albums['All Photos']
 
 
+class PhotosService(PhotoLibrary):
+    """The 'Photos' iCloud service.
+
+    This also acts as a way to access the user's primary library.
+    """
+    def __init__(self, service_root, session, params):
+        self.session = session
+        self.params = dict(params)
+        self._service_root = service_root
+        self._service_endpoint = \
+            ('%s/database/1/com.apple.photos.cloud/production/private'
+             % self._service_root)
+
+        self._libraries = None
+
+        self.params.update({
+            'remapEnums': True,
+            'getCurrentSyncToken': True
+        })
+
+        # TODO: Does syncToken ever change?
+        # self.params.update({
+        #     'syncToken': response['syncToken'],
+        #     'clientInstanceId': self.params.pop('clientId')
+        # })
+
+        self._photo_assets = {}
+
+        super(PhotosService, self).__init__(
+            service=self, zone_id={u'zoneName': u'PrimarySync'})
+
+    @property
+    def libraries(self):
+        if not self._libraries:
+            url = ('%s/changes/database' %
```

Comment:
I got this working again by using

```
            url = ('%s/zones/list' %
```
            
here.

Reply:
independent of this, it should also work if one manually looks for the correct library name on iCloud.com when showing the fotos manually and giving that to the --library parameter

Reply:
How do you find the shared library name on icloud.com? Thank you.

Reply:
open photos on iCloud.com and watch the network requests using developer mode.
Look for the zones/list call.
The shared library name is in the result json of this call.